### PR TITLE
[#156] splitting azure into non-wildfly and wildfly module

### PR DESCRIPTION
**Pull request by istraka**
For #156 

@mchoma @rsvoboda

Key changes:
- `azure` module split into `azure` and `azure-wildfly`. The latter one is extending the first one bringing additional support for injecting `OnlineManagementClient` and deploying Archive (this operation is product specific).
- `core` module split into `core` and `core-wildfly`. The later one brings WildFly specific annotations and dependencies (creaper , ...)
- Injecting and deploying services (SPI) redesigned.
- AWS is out of scope right now (temporarily removed from pom).

Examples:
- Azure WEB APP archive deployment (**WildFly**)
  - You can use `WildFly` annotation. If not present, default one will be used. This principle is used also for injecting `OnlineManagementClient`.
    - https://github.com/istraka/sunstone/blob/issue_%23156/core-wildfly/src/main/java/sunstone/annotation/WildFly.java#L19
  - https://github.com/istraka/sunstone/blob/issue_%23156/azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/suitetests/AzureWebAppDeployFirstTest.java
  - Azure WEB APP resource creation & hostname injection (**non-WildFly**)
    - It is EAP web app, but container can be changed here
      - https://github.com/istraka/sunstone/blob/issue_%23156/azure/src/test/resources/sunstone/azure/armTemplates/eapWebApp.json#L11 
    - https://github.com/istraka/sunstone/blob/issue_%23156/azure/src/test/java/sunstone/azure/armTemplates/di/AzureWebAppTests.java

### CODE REVIEW

File: azure/src/main/java/sunstone/azure/impl/AzureSunstoneResourceInjectorProvider.java [hunk 1]
```diff
@@ -1,23 +1,43 @@
...
 import java.lang.reflect.Field;
+import java.util.List;
 import java.util.Optional;
 
 /**
  * Used by {@link sunstone.core.SunstoneExtension} to get {@link AzureSunstoneResourceInjector} if it can handle the annotation
  * and the type (clazz) of field of injection operation.
  */
 public class AzureSunstoneResourceInjectorProvider implements SunstoneResourceInjectorProvider {
+//    @Override
```
Comment (mchoma) [inline azure/src/main/java/sunstone/azure/impl/AzureSunstoneResourceInjectorProvider.java:19]: apparently some leftover?

File: core/src/main/java/sunstone/core/api/SunstoneArchiveDeployer.java [hunk 2]
```diff
@@ -3,12 +3,10 @@
 import org.junit.jupiter.api.extension.ExtensionContext;
 import sunstone.core.exceptions.SunstoneException;
 
-import java.io.InputStream;
-import java.lang.annotation.Annotation;
-
 /**
- *
+ * // todo
```
Comment (mchoma) [inline core/src/main/java/sunstone/core/api/SunstoneArchiveDeployer.java:7]: todo :)

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/di/AzStandaloneManagementClientTests.java [hunk 3]
```diff
@@ -22,19 +22,22 @@
...
-        template = "sunstone/azure/armTemplates/eap.json", group = SunstoneResourceAzSuiteTests.group, perSuite = true)
+        template = "sunstone/azure/armTemplates/eap.json", group = AzStandaloneManagementClientTests.group, perSuite = true)
 public class AzStandaloneManagementClientTests {
+    public static final String group = "${azure.group:sunstone-testing-group}";
 
     @AzureVirtualMachine(name = AzureTestConstants.instanceName)
     static OnlineManagementClient staticMngmtClient;
 
-    @AzureVirtualMachine(name = AzureTestConstants.instanceName, group = SunstoneResourceAzSuiteTests.group, mode = OperatingMode.STANDALONE, standalone = @StandaloneMode(user = AzureTestConstants.mngmtUser, password = AzureTestConstants.mngmtPassword, port = AzureTestConstants.mngmtPort))
+    @AzureVirtualMachine(name = AzureTestConstants.instanceName, group = AzStandaloneManagementClientTests.group)
```
Comment (mchoma) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/di/AzStandaloneManagementClientTests.java:32]: One can even say, having it separated like this is even cleaner/nicer. Btw. haven't we discussed this option in past?

But we can see (in this MR) it comes with bigger complexity/work. Now question is if it is worth in long term.

File: azure/src/main/java/sunstone/azure/impl/AzureArchiveDeployerProvider.java [hunk 4]
```diff
@@ -1,26 +0,0 @@
...
-import sunstone.azure.annotation.AzureArchiveDeploymentAnnotation;
-import sunstone.core.AnnotationUtils;
-import sunstone.core.api.SunstoneArchiveDeployer;
-import sunstone.core.spi.SunstoneArchiveDeployerProvider;
-
-import java.lang.annotation.Annotation;
-import java.util.Optional;
-
-
-public class AzureArchiveDeployerProvider implements SunstoneArchiveDeployerProvider {
```
Comment (mchoma) [inline azure/src/main/java/sunstone/azure/impl/AzureArchiveDeployerProvider.java:14]: Just idea. Isn't product agnostic way of deployment copying war/jar "somewhere" on azure. I saw sunstone is already dealing with ftp in some cases. Can't that be used?

Comment (istraka) [reply azure/src/main/java/sunstone/azure/impl/AzureArchiveDeployerProvider.java:14]: It depends on the resource. On VMs, there is no universal way. On App Services, partially yes, you can specify a gith repository. On the other hand appservice.deploy() (from Azure SDK) supports only WAR archive type for EAP stack and not JAR, nor EAR.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/di/AzStandaloneManagementClientTests.java (hunk 3 above)
Comment (istraka) [reply azure-wildfly/src/test/java/sunstone/azure/armTemplates/di/AzStandaloneManagementClientTests.java:32]: We might have:)

Yeah, te project is more complex. Modules kinda extending other modules and depending on them, not really something that is widely used. However, the vision is to enable cross team collaboration. How much does it worth it is yet to be determined.

**Comment by istraka** [comment]
Merging...

**Pull request merged**